include/termtools.h

```diff
@@ -56,7 +56,44 @@
 #include <stdbool.h>
 #include <locale.h>
 
-/*
+/**
+ * sets the locale to unicode.
+ * ---------------------------------------------------------------
+ * first, sets the locale for the application to the system locale
+ * defined in Windows, rater than the default "C". after this, it
+ * will set the locale to codepage "UTF-8" (65001).
+ * 
+ * Example: C -> de_DE.1252 -> de_DE.UTF-8
+ * 
+ * if the correct locale can not be determined, it will be set
+ * to "en_US.UTF-8".
+ * ---------------------------------------------------------------
+ */
+void setUnicodeLocale()
+{
+    LPWSTR pbOldLocale = _wsetlocale(LC_ALL, L"");
+    LPWSTR pbTmp = NULL, pbTokContext = NULL;
+
+    pbTmp = wcstok_s(pbOldLocale, L".", &pbTokContext);
+
+    if(pbTmp && _wcsicmp(pbTmp, L"C") != 0)
+    {
+        LPWSTR pbLocaleUnicode = NULL;
+        SIZE_T cbLocaleUnicode = (sizeof(WCHAR) * wcslen(pbTmp)) + 7;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Does not handle strings that are not \0-terminated; if given one it may perform an over-read (it could cause a crash if unprotected) (CWE-126).  ](https://app.codacy.com/manual/HolgerDoerner/TermTools/pullRequest?prid=4765330)